python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -501,6 +501,14 @@ def test_float_nulls(self):
         result = table.to_pandas()
         tm.assert_frame_equal(result, ex_frame)
 
+    def test_float_nulls_to_ints(self):
+        # ARROW-2135
+        df = pd.DataFrame({"a": [1.0, 2.0, pd.np.NaN]})
+        schema = pa.schema([pa.field("a", pa.int16(), nullable=True)])
+        table = pa.Table.from_pandas(df, schema=schema)
+        assert table[0].to_pylist() == [1, 2, None]
+        tm.assert_frame_equal(df, table.to_pandas())
```

Comment:
Is there already a test for things like `a = [1.0, 2.0, 3.1, np.nan]` where a user passes in an integer type?

Reply:
You mean for the truncation behavior? Let me look.

Reply:
No, I don't think so. I'm not sure we specify the truncation mode anywhere either?

Reply:
It looks like it's a hard cast:

```
In [7]: pa.array([1, 2, 3.190, np.nan], type=pa.int64())
Out[6]:
<pyarrow.lib.Int64Array object at 0x7f537e42dd68>
[
  1,
  2,
  3,
  NA
]
```

Reply:
That's fine. Was just wondering.